tests/app_test.py

```diff
@@ -88,65 +33,169 @@ def test_find_valid_refresh_token(test_user, db_session):
+    client, persisted_refresh_tokens, auth_header
 ):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
     res = client.get("/token/", headers=auth_header)
     assert res.status_code == 403
 
 
-def test_token_endpoint_with_bogus_idp(client, test_user, db_session, auth_header):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
-    # make sure the IDP we use is "default" when no IDP is requested
+def test_token_endpoint_with_bogus_idp(client, persisted_refresh_tokens, auth_header):
     res = client.get("/token/?idp=bogus", headers=auth_header)
     assert res.status_code == 400
 
 
-def test_token_endpoint_without_auth_header(client, test_user, db_session):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
+def test_token_endpoint_without_auth_header(client, persisted_refresh_tokens):
     res = client.get("/token/")
     assert res.status_code == 403
 
 
+def assert_authz_mapping_for_test_user_in_default_commons(authz_mapping):
+    assert "/1" in authz_mapping
+    assert "/2" not in authz_mapping
+    assert "/3" not in authz_mapping
+    assert "/4" not in authz_mapping
+    assert "/5" not in authz_mapping
+
+
+def assert_authz_mapping_for_test_user_in_idp_a_commons(authz_mapping):
+    assert "/3" in authz_mapping
+    assert "/1" not in authz_mapping
+    assert "/2" not in authz_mapping
+    assert "/4" not in authz_mapping
+    assert "/5" not in authz_mapping
+
+
+def test_aggregate_user_user_endpoint(
+    app, client, persisted_refresh_tokens, auth_header
+):
+    res = client.get("/aggregate/user/user", headers=auth_header)
+    assert res.status_code == 200
+    assert len(res.json) == 2
+
+    default_commons_hostname = app.config["OIDC"]["default"]["commons_hostname"]
+    assert default_commons_hostname in res.json
+    assert len(res.json[default_commons_hostname]) == 3
+    assert_authz_mapping_for_test_user_in_default_commons(
+        res.json[default_commons_hostname]["authz"]
+    )
+
+    idp_a_commons_hostname = app.config["OIDC"]["idp_a"]["commons_hostname"]
+    assert idp_a_commons_hostname in res.json
+    assert len(res.json[idp_a_commons_hostname]) == 3
+    assert_authz_mapping_for_test_user_in_idp_a_commons(
+        res.json[idp_a_commons_hostname]["authz"]
+    )
+
+
+def test_aggregate_user_user_endpoint_with_filters(
+    app, client, persisted_refresh_tokens, auth_header
+):
+    res = client.get(
+        "/aggregate/user/user?filters=authz&filters=role", headers=auth_header
+    )
+    assert res.status_code == 200
+    assert len(res.json) == 2
+
+    default_commons_hostname = app.config["OIDC"]["default"]["commons_hostname"]
+    assert default_commons_hostname in res.json
+    assert len(res.json[default_commons_hostname]) == 2
+    assert "role" in res.json[default_commons_hostname]
```

Comment:
should we also check that _only_ `authz` and `role` are returned?

Reply:
This should be covered now because `assert len(res.json[default_commons_hostname]) == 2` and then `authz` and `role` are both checked.